patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -68,7 +437,220 @@
  			}
  
  			if (x < roomX1)
-@@ -2273,10 +_,20 @@
+@@ -2121,6 +_,8 @@
+ 		}
+ 
+ 		public static bool meteor(int i, int j, bool ignorePlayers = false) {
++			var meteorTasks = new List<GenPass>();
++
+ 			if (i < 50 || i > Main.maxTilesX - 50)
+ 				return false;
+ 
+@@ -2163,101 +_,110 @@
+ 				}
+ 			}
+ 
++			meteorTasks.Add(new PassLegacy("Meteor Spawn", (progress, config) => {
++				num = genRand.Next(17, 23);
++				for (int num2 = i - num; num2 < i + num; num2++) {
++					for (int num3 = j - num; num3 < j + num; num3++) {
++						if (num3 <= j + Main.rand.Next(-2, 3) - 5)
```

Comment:
This is a terrible patch. Split the method into 3. Use the middle one as a delegate.